lib/readline.js

```diff
@@ -90,6 +90,12 @@ function Interface(input, output, completer, terminal) {
   this._previousKey = null;
   this.escapeCodeTimeout = ESCAPE_CODE_TIMEOUT;
 
+  // stuff used for reverse search
+  this.reverseSearchPrompt = "(reverse-i-search)`' ";
+  this.inReverseSearch = false;
+  this.reverseSearchResults = [];
+  this.reverseSearchIndex = 0;
```

Comment:
I might be wrong but I believe it is possible to combine `reverseSearchIndex` and `inReverseSearch` by setting `reverseSearchIndex` to `-1` when not searching.